shared/tests/APIv3.test.ts

```diff
@@ -0,0 +1,149 @@
+    seatsFilled: 28,
+    potentialError: false,
+    startDate: {
+        year: 2023,
+        month: 1,
+        day: 17,
+    },
+    endDate: {
+        year: 2023,
+        month: 5,
+        day: 12,
+    },
+    permCount: 2,
+} as APIv4.Section;
+
+const v3Course: APIv3.Course = {
+    courseCode: "CSCI 105 HM-01",
+    courseName: "Computer Systems",
+    courseSortKey: ["CSCI 105 HM-01"],
+    courseMutualExclusionKey: ["CSCI 105 HM"],
+    courseDescription:
+        "An introduction to computer systems. In particular the course investigates data representations, machine level representations of programs, processor architecture, program optimizations, the memory hierarchy, linking, exceptional control flow (exceptions, interrupts, processes and Unix signals), performance measurement, virtual memory, system-level I/O and basic concurrent programming. These concepts are supported by a series of hands-on lab assignments. Prerequisite: Computer Science 70.",
+    courseInstructors: ["Geoffrey Kuenning"],
+    courseTerm: "SP2023",
+    courseSchedule: [
+        {
+            scheduleDays: "F",
+            scheduleStartTime: "13:15",
+            scheduleEndTime: "14:30",
+            scheduleStartDate: "2023-01-17",
+            scheduleEndDate: "2023-05-12",
+            scheduleTermCount: 1,
+            scheduleTerms: [0],
+            scheduleLocation: "McGregor CompSci Center 203",
+        },
+        {
+            scheduleDays: "F",
+            scheduleStartTime: "13:15",
+            scheduleEndTime: "14:30",
+            scheduleStartDate: "2023-01-17",
+            scheduleEndDate: "2023-05-12",
+            scheduleTermCount: 1,
+            scheduleTerms: [0],
+            scheduleLocation: "McGregor CompSci Center 204",
+        },
+        {
+            scheduleDays: "F",
+            scheduleStartTime: "13:15",
+            scheduleEndTime: "14:30",
+            scheduleStartDate: "2023-01-17",
+            scheduleEndDate: "2023-05-12",
+            scheduleTermCount: 1,
+            scheduleTerms: [0],
+            scheduleLocation: "McGregor CompSci Center 205",
+        },
+        {
+            scheduleDays: "TR",
+            scheduleStartTime: "13:15",
+            scheduleEndTime: "14:30",
+            scheduleStartDate: "2023-01-17",
+            scheduleEndDate: "2023-05-12",
+            scheduleTermCount: 1,
+            scheduleTerms: [0],
+            scheduleLocation: "Beckman Hall B126",
+        },
+    ].sort(compareSchedules),
+    courseCredits: 3,
+    courseSeatsTotal: 30,
+    courseSeatsFilled: 28,
+    courseWaitlistLength: null,
+    courseEnrollmentStatus: "reopened",
+    permCount: 2,
+};
+
+function compareSchedules(x: APIv3.Schedule, y: APIv3.Schedule): number {
+    const xString = JSON.stringify(x);
+    const yString = JSON.stringify(y);
+    if (xString < yString) {
+        return -1;
+    } else if (xString > yString) {
+        return +1;
+    } else {
+        return 0;
+    }
```

Comment:
```typescript
return JSON.stringify(x).localeCompare(JSON.stringify(y))
```

Reply:
Thanks! I am gonna keep `stringifyWeekdayEnumArray` as a separate function just so the code is more readable, but I will redo everything else in correspondence with your suggestions.